src/platform/Darwin/DnssdContexts.cpp

```diff
@@ -607,7 +607,7 @@ bool ResolveContext::TryReportingResultsForInterfaceIndex(uint32_t interfaceInde
     {
         auto delegate = static_cast<DiscoverNodeDelegate *>(context);
         DiscoveredNodeData nodeData;
-        service.ToDiscoveredNodeData(addresses, nodeData);
+        service.ToDiscoveredCommissionNodeData(addresses, nodeData);
```

Comment:
So this does not make much sense to me.  This could be an operational browse, no?  `DiscoverNodeDelegate` claims to support operational, commissionable, etc.

I think we are fundamentally making an odd assumption that "operational browse means just finding the instance name" or something like that, but that's not what DiscoverNodeDelegate consumers want or get....

Reply:
I was not sure if Darwin supports operational browse yet. I might have incorrectly assumed that it doesn't. 
We ideally need to call either `ToDiscoveredCommissionNodeData()` or `ToDiscoveredOperationalNodeBrowseData()` based on the browse type. This is what I have done in `Discovery_ImplPlatform.cpp`.
Does that make sense?

Reply:
> I was not sure if Darwin supports operational browse yet.

I am telling you that it does.

And what it expects out of an operational browse in this case is complete information including the IPs and whatnot, no?

Reply:
Well, my point is that there is existing operational browse code, and that code is expected to produce IPs and whatnot, and it's used by other code...

Maybe that means we need to figure out what our data structures and their naming look like better, but that ResolveContext code is absolutely reached for any kind of browse, and DiscoverNodeDelegate explicitly supports any kind of browse and produces full information for it.

If we have to do things this way for now just to unblock you we can, but the naming is pretty broken and we should at least have a followup to fix it...

Reply:
`DiscoverNodeDelegate` previously only delivered `CommissionNodeData`, even if the darwin implementation supported operational browse, the user of `DiscoverNodeDelegate` would probably not be expecting it. Also, during the refactor we have made sure all existing users of `DiscoverNodeDelegate` check that `DiscoverNodeData` has `CommissionNodeData` instead of `OperationalNodeBrowseData`.

Anyways, I will need this to get unblocked. So, please suggest how to proceed. Do I need to have check for serviceType and fill the right structure (i.e. either `CommissionNodeData` or `OperationalNodeBrowseData`) here? or does the current changes looks good for now?

Reply:
As discussed on slack, have added check for serviceType to fill the right structure before calling delegate.